drivers/gles3/shaders/canvas.glsl

```diff
@@ -278,6 +278,18 @@ float map_ninepatch_axis(float pixel, float draw_size, float tex_pixel_size, flo
 			ratio = mod(ratio * scale, 1.0);
 			// Scale to source texture.
 			return (margin_begin + ratio * dst_area) * tex_pixel_size;
+		} else if (np_repeat == 3) { // Center.
+			float src_area = draw_size - margin_begin - margin_end;
+			float dst_area = tex_size - margin_begin - margin_end;
+			float center = (src_area - dst_area) / 2.0;
+			if (pixel <= margin_begin + center) {
+				return (margin_begin + 1) * tex_pixel_size; // Stretch begin.
+			}
+			if (pixel >= draw_size - margin_end - center) {
+				return (tex_size - margin_end - 1) * tex_pixel_size; // Stretch end.
```

Comment:
The `+ 1` / `- 1` parts makes the streching be not continous. Here's an example of this in action:

||||
|--------|--------|--------|
|![godot windows editor dev x86_64_Em59ahuKlT](https://github.com/godotengine/godot/assets/9283098/e0b2fa0d-086b-4b74-bf44-4a3d07a4beef)|![godot windows editor dev x86_64_1DSFjZecrP](https://github.com/godotengine/godot/assets/9283098/a24eb359-74fc-4879-8f81-10be91b74810)|![C7rtBPqk4F](https://github.com/godotengine/godot/assets/9283098/fff414e7-456a-46d8-84dd-c4dc71053c86)|


The left margin is set so the center part starts at the rightmost pixel of the nose. The left margin is rendered as expected, same center, but the stretched part between them samples the pixel to the right of the nose (because of the `+ 1`), creating discontinuity.

Yeah, this looks wrong:
![61ohEJvqJ9](https://github.com/godotengine/godot/assets/9283098/62eab3ac-361d-44c7-ac10-4cdeae8e898c)